src/main/java/org/opensearch/knn/index/query/BaseQueryFactory.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.query;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.extern.log4j.Log4j2;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.join.BitSetProducer;
+import org.apache.lucene.search.join.ToChildBlockJoinQuery;
+import org.opensearch.index.query.QueryBuilder;
+import org.opensearch.index.query.QueryShardContext;
+import org.opensearch.index.search.NestedHelper;
+import org.opensearch.knn.index.VectorDataType;
+import org.opensearch.knn.index.util.KNNEngine;
+
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+* Base class for creating vector search queries.
+*/
+@Log4j2
+public abstract class BaseQueryFactory {
```

Comment:
i think what you need is a Single Factory class or method that will return KNNQuery ( instance of Radial or KNN based on parameter )

Reply:
The motivation of this refactor is come from https://github.com/opensearch-project/k-NN/pull/1498#discussion_r1514903363